editor/components/post-text-editor/index.js

```diff
@@ -18,41 +18,45 @@ import { getEditedPostContent } from '../../store/selectors';
 import { editPost, resetBlocks, checkTemplateValidity } from '../../store/actions';
 
 class PostTextEditor extends Component {
-	constructor( props ) {
+	constructor() {
 		super( ...arguments );
 
-		this.onChange = this.onChange.bind( this );
-		this.onPersist = this.onPersist.bind( this );
+		this.handleFocus = this.handleFocus.bind( this );
+		this.handleChange = this.handleChange.bind( this );
+		this.handleBlur = this.handleBlur.bind( this );
 
 		this.state = {
-			initialValue: props.value,
+			value: null,
+			isDirty: false,
 		};
 	}
 
-	onChange( event ) {
-		this.props.onChange( event.target.value );
+	handleFocus() {
```

Comment:
FWIW I find these sorts of function names aren't very informative and encourage developers to not clearly separate the responsibilities of functions. As an outsider looking at this function later, I'd have no idea what it's intending to do, at least not without reading through and interpreting the logic (and who has time for that?). If it were named `setStateValue` with a DocBlock explaining how we track a local copy of the value to avoid clobbering edits from global state, I'd feel much more informed.

There's a few places where I've tried to set more explicit patterns for some of these common behaviors. One in particular is handling `onKeyDown`, since we usually want distinct functionality to take place on individual key codes. While maybe a bit extreme (especially since it only handles as a single key code _at the moment_), `NavigableToolbar` is one such example:

https://github.com/WordPress/gutenberg/blob/4bac4fe5e4f918d1ba24ee12e8f00f6fcc89b49c/editor/components/navigable-toolbar/index.js#L32-L34

https://github.com/WordPress/gutenberg/blob/4bac4fe5e4f918d1ba24ee12e8f00f6fcc89b49c/editor/components/navigable-toolbar/index.js#L51-L73

https://github.com/WordPress/gutenberg/blob/4bac4fe5e4f918d1ba24ee12e8f00f6fcc89b49c/editor/components/navigable-toolbar/index.js#L83

(And yes, arguments can be made that `_.cond` is not a very obvious / readable utility, but better abstractions could exist. The general point stands all the same)

Reply:
👍 noted. I've moved away from 'handle-' method names in c4035be7a.